src/theory/arrays/theory_arrays_rewriter.h

```diff
@@ -371,7 +371,7 @@ class TheoryArraysRewriter {
               elements.pop_back();
             }
             Trace("arrays-postrewrite") << "Arrays::postRewrite returning " << n << std::endl;
-            return RewriteResponse(REWRITE_DONE, n);
+            return RewriteResponse((n == node) ? REWRITE_DONE : REWRITE_AGAIN, n);
```

Comment:
Actually, after rereading the code (which is pretty obscure, sorry), I think that in this case it shouldn't ever be possible for n == node to be true.  That's because we know we switched at least two indices when we are in this case.  So, I think we should just always call REWRITE_AGAIN.